packages/remix-dev/vite/plugin.ts

```diff
@@ -404,6 +404,12 @@ export const remixVitePlugin: RemixVitePlugin = (options = {}) => {
         return {
           appType: "custom",
           experimental: { hmrPartialAccept: true },
+          optimizeDeps: {
+            include: ["react", "react-dom/client"],
```

Comment:
should probably look into adding JSX import runtimes to `include` like in `vite-plugin-react`